packages/legend-graph/src/graphManager/protocol/pure/v1/transformation/pureGraph/to/helpers/V1_DomainBuilderHelper.ts

```diff
@@ -262,6 +264,23 @@ export const V1_buildAssociationProperty = (
     associatedProperty.type,
     `Association associated property 'type' field is missing`,
   );
+  const _class = context.graph.getMainGraphClass(associatedPropertyClassType);
+  const _association = context.graph.getOwnNullableAssociation(
+    pureAssociation.path,
+  );
+  if (!_class && _association) {
+    const message = `Found dependency class property '${associatedPropertyClassType}' in association '${pureAssociation.path}'`;
+    /**
+     * In strict-mode, graph builder will consider this as an error
+     * See https://github.com/finos/legend-studio/issues/941
+     *
+     * @discrepancy graph-building
+     */
+    if (context.options?.strict) {
+      throw new GraphBuilderError(message);
+    }
+    context.log.warn(LogEvent.create(message));
+  }
   const associatedClass = context.resolveClass(associatedPropertyClassType);
```

Comment:
Can't we move this logic as a helper method. takes the association, the property , pure model. Then check if assoication, property class are in the main graph.